### PR TITLE
Issue 118 fill transfers bad debt

### PR DESCRIPTION
**Pull request by markuspluna**

### CODE REVIEW

File: lending-pool/src/auctions/bad_debt_auction.rs [hunk 1]
```diff
@@ -82,6 +82,13 @@ pub fn fill_bad_debt_auction(
         &filler_state.address,
     );
 
+    // If the backstop still has liabilities and less than 10% of the backstop threshold burn bad debt
+    if backstop_state.positions.liabilities.len() > 0
+            //TODO: this token check needs to check k-value of pool balance LP tokens
+        && backstop_client.pool_balance(&e.current_contract_address()).tokens < 100_000_0000
```
Comment (mootz12) [inline lending-pool/src/auctions/bad_debt_auction.rs:88]: does 100k represent the kValue or representative Blend Tokens? I'm gonna assume the former

Comment (mootz12) [reply lending-pool/src/auctions/bad_debt_auction.rs:88]: Think the underscores confused me, this is 100 tokens at the moment?

Do we have the k-value threshold we are going to use for the pool balance figured out?

File: lending-pool/src/pool/bad_debt.rs [hunk 2]
```diff
@@ -30,7 +15,8 @@ pub fn manage_bad_debt(e: &Env, user: &Address) {
 ///
 /// ### Panics
 /// If the user does not have bad debt
-fn transfer_bad_debt_to_backstop(e: &Env, user: &Address, backstop: &Address) {
+//TODO: this could be used to transfer backstop bad debt to itself - should we prevent this?
```
Comment (mootz12) [inline lending-pool/src/pool/bad_debt.rs:18]: verifying that user !== backstop feels appropriate

- [ ] do the TODO

File: lending-pool/src/pool/bad_debt.rs [hunk 3]
```diff
@@ -61,49 +48,28 @@ fn transfer_bad_debt_to_backstop(e: &Env, user: &Address, backstop: &Address) {
     new_user_state.store(e);
 }
 
-/// Burn bad debt from the backstop. This can only occur if the backstop module has reached a critical balance and
-/// a bad debt auction is not currently occurring.
+/// Burn bad debt from the backstop. This can only occur if the backstop module has reached a critical balance
 ///
```
Comment (mootz12) [inline lending-pool/src/pool/bad_debt.rs:52]: additional code block line

**Review by mootz12 — CHANGES_REQUESTED** [review]
Looks good! Had one question and I think the TODO you called out should be taken care of before merging

File: lending-pool/src/auctions/bad_debt_auction.rs (hunk 1 above)
Comment (markuspluna) [reply lending-pool/src/auctions/bad_debt_auction.rs:88]: yeah threshold is 200,000 based on the fact that it costs around 10k to spin up a pool at 1.2 cent blend and 50k at 10 cent blend

I'll update the amounts to reflect that

File: lending-pool/src/pool/bad_debt.rs [hunk 4]
```diff
@@ -245,67 +246,4 @@ mod tests {
             transfer_bad_debt_to_backstop(&e, &samwise);
         });
     }
-
-    /***** burn_backstop_bad_debt ******/
-
-    #[test]
-    fn test_burn_backstop_bad_debt() {
```
Comment (mootz12) [inline lending-pool/src/pool/bad_debt.rs:286]: was this test removal an accident? It appears to be the only one that tests `burn_backstop_bad_debt`

File: lending-pool/src/pool/bad_debt.rs [hunk 5]
```diff
@@ -27,6 +26,9 @@ pub fn transfer_bad_debt_to_backstop(e: &Env, user: &Address) {
     let mut pool = Pool::load(e);
     let reserve_list = storage::get_res_list(e);
     let backstop_address = storage::get_backstop(e);
+    if user.clone() == backstop_address {
+        panic_with_error!(e, PoolError::BadRequest);
+    }
```
Comment (mootz12) [inline lending-pool/src/pool/bad_debt.rs:31]: can a test be added to exercise this?

Also, we should validate this before loading a bunch of data incurring a cost to the caller.

**Review by mootz12 — CHANGES_REQUESTED** [review]
Looks like a necessary unit test was removed, and an additional one is needed to exercise the new `backstop_address` validation

**Review by mootz12 — APPROVED** [review]

**Pull request merged**